atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/bucketingthings/SweepBucketRecordsTable.java

```diff
@@ -17,13 +17,14 @@
 package com.palantir.atlasdb.sweep.asts.bucketingthings;
 
 import com.palantir.atlasdb.sweep.asts.TimestampRange;
+import java.util.Optional;
 
 public interface SweepBucketRecordsTable {
     /**
-     * Returns the {@link TimestampRange} for the given bucket identifier, throwing a
-     * {@link java.util.NoSuchElementException} if one is not present.
+     * Returns a {@link TimestampRange} for the given bucket identifier, if one exists. Iff a bucket is closed, then
+     * the corresponding record will be present. (If the bucket is open, no record will be present.)
```

Comment:
This isn't atomic though right? The bucket writer, in order, does the following:

1. writes closed buckets to the buckets table
2. writes the bucket record
3. sets the state machine to be ready going in the next state

So if we happen to launch this query between steps 1 and 2, then there could be a closed bucket with no record in the bucket records table, unless I've misunderstood something. It's more "if record present, bucket definitively closed - if record not present, bucket could be open or closed".

For what it's worth I think your prototype in #7424 handles this acceptably, though I'm not fully certain if the logging there is always correct - it is OK to end the bucket probe even if the bucket was closed (and we're only doing it for an edge case where transitioning to ready keeps failing).